plugins/inputs/intel_powerstat/file.go

```diff
@@ -133,19 +133,34 @@ func (fs *fileServiceImpl) readFileToFloat64(reader io.Reader) (float64, int64,
 }
 
 // readFileAtOffsetToUint64 reads 8 bytes from passed file at given offset.
-func (fs *fileServiceImpl) readFileAtOffsetToUint64(reader io.ReaderAt, offset int64) (uint64, error) {
+func (fs *fileServiceImpl) readFileAtOffsetToUint64(reader io.ReaderAt, offset int64, timeout time.Duration) (uint64, error) {
 	buffer := make([]byte, 8)
 
 	if offset == 0 {
 		return 0, fmt.Errorf("file offset %d should not be 0", offset)
 	}
 
-	_, err := reader.ReadAt(buffer, offset)
-	if err != nil {
+	value := make(chan uint64)
+	errs := make(chan error)
+
+	go func() {
+		if n, readErr := reader.ReadAt(buffer, offset); n >= 0 {
+			value <- binary.LittleEndian.Uint64(buffer)
+		} else {
+			errs <- readErr
+		}
```

Comment:
It may happen that `reader.ReadAt(buffer, offset)` returns `n` bytes read where `n < len(buffer)` AND non-nil `err`. For this case read value is not valid and return should be returned.
In your implementation it would be handled as a properly read value.